### PR TITLE
Rename the Controller::reset() method

### PR DESCRIPTION
**Pull request by aschempp**
Fixes https://github.com/contao/contao/issues/4257

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Controller.php [hunk 1]
```diff
@@ -1450,7 +1450,11 @@ public static function loadDataContainer($strTable, $blnNoCache=false)
 		$loader->load($blnNoCache);
 	}
 
-	public static function reset()
+	/**
+	 * Do not name this "reset" because it might result in conflicts with child classes
+	 * @see https://github.com/contao/contao/issues/4257
+	 */
+	public static function resetStaticCache()
```
Comment (Toflar) [inline core-bundle/src/Resources/contao/library/Contao/Controller.php]: `@internal` maybe?

Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: I thought about that as well, but the we should make all our reset methods `@internal`? And maybe rename them to `resetStaticCache`?

**Review by ausi — DISMISSED** [review]

**Review by Toflar — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Controller.php (hunk 1 above)
Comment (Toflar) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: I think we can discuss that at the devs meeting real quick. @leofeyer can you put that on the list? Should be a matter of a few minutes.

**Comment by m-vo** [comment]
I get the intention, but now we're renaming a public method that is not internal. That doesn't feel right.

**Comment by Toflar** [comment]
It's only been introduced in the latest version. It can be considered a bugfix imho.

File: core-bundle/src/Resources/contao/library/Contao/Controller.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: Done. ✔️

**Review by ausi — DISMISSED** [review]

**Review by ausi — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

**Pull request merged**